src/api/app/controllers/services/webhooks_controller.rb

```diff
@@ -0,0 +1,46 @@
+class Services::WebhooksController < ApplicationController
+  #
+  # This controller is checking permission always only on the base of the github signature
+  #
+  skip_before_action :extract_user
+  skip_before_action :require_login
+  skip_before_action :validate_params
+  before_action :validate_token, :set_package
+
+  def create
+    if !@user.is_active? || !@user.can_modify?(@package)
+      render_error message: 'Token not found or not valid.', status: 404
+      return
+    end
+
+    Backend::Api::Sources::Package.trigger_services(@project.name, @package.name, @user.login)
+    render_ok
+  end
+
+  private
+
+  def set_package
+    @package = @token.package || Package.get_by_project_and_name(params[:project], params[:package], use_source: true)
+    @project = @package.project
+  end
+
+  def validate_token
+    @token = Token::Service.find_by(id: params[:id])
+    if !@token || !@token.valid_signature?(signature, payload_body)
+      render_error message: 'Token not found or not valid.', status: 403
+      return false
+    end
+    @user = @token.user
```

Comment:
:thought_balloon: I find this indirection rather confusing. Can you please use @token.user directly? Or do `set_user` like you do `set`package`.